autoscheduler/frontend/src/tests/redux/CourseCards.test.ts

```diff
@@ -274,4 +284,94 @@ describe('Course Cards Redux', () => {
       });
     });
   });
+
+  describe('clearCourseCards', () => {
+    test('deletes all course cards and resets numCardsCreated', () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+
+      // act
+      store.dispatch(clearCourseCards());
+
+      // assert
+      expect(store.getState().courseCards).toEqual({ numCardsCreated: 0 });
+    });
+  });
+
+  describe('replaceCourseCards', () => {
+    test('replaces all course cards and keeps all properties not related to sections', async () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      const expectedCourseCards: CourseCardArray = {
+        0: {
+          course: 'MATH 151',
+          customizationLevel: CustomizationLevel.BASIC,
+        },
+        numCardsCreated: 1,
+      };
+      const courseCards = [
+        {
+          course: 'MATH 151',
+          customizationLevel: CustomizationLevel.BASIC,
+          web: 'no_preference',
+          honors: 'exclude',
+        },
+      ];
+      fetchMock.mockImplementationOnce(testFetch);
+
+      // act
+      store.dispatch<any>(replaceCourseCards(courseCards, '202031'));
+      // wait for all actions to finish
+      await new Promise(setImmediate);
+
+      // assert
+      expect(store.getState().courseCards).toMatchObject(expectedCourseCards);
+    });
+
+    test('adds non-existing sections for course cards', async () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      const courseCards = [
+        {
+          course: 'MATH 151',
+          customizationLevel: CustomizationLevel.BASIC,
+        },
+      ];
+      fetchMock.mockImplementationOnce(testFetch);
+
+      // act
+      store.dispatch<any>(replaceCourseCards(courseCards, '202031'));
+      // wait for all actions to finish
+      await new Promise(setImmediate);
+
+      // assert
+      // testFetch with a MATH course has 1 section, which should be added
+      expect(store.getState().courseCards[0].sections).toHaveLength(1);
+    });
+
+    test('keeps selected sections from courseCards', async () => {
+      // arrange
+      const store = createStore(autoSchedulerReducer, applyMiddleware(thunk));
+      // section is the id of the section returned by testFetch, should be checked
+      const section = 830262;
+      fetchMock.mockImplementationOnce(testFetch);
+
+      const courseCards = [
```

Comment:
This is the serialized version, right?

Reply:
Yeah, I should declare this as `SerializedCourseCardOptions[]` to make it more clear